src/gtt_log.c

```diff
@@ -77,13 +87,49 @@ log_write (time_t t, const char *logstr)
     strcpy (date, "???");
 
   /* Append to end of file */
-  gnome_vfs_seek (handle, GNOME_VFS_SEEK_END, 0);
-  GnomeVFSFileSize bytes_written;
-  gnome_vfs_write (handle, date, strlen (date), &bytes_written);
-  gnome_vfs_write (handle, logstr, strlen (logstr), &bytes_written);
-  gnome_vfs_write (handle, "\n", 1, &bytes_written);
```

Comment:
So this is an example of "code expansion" you've replaced six lines of code by 40, and even invented a new struct along the way. The code is mostly a cut-and-paste of exactly the same stuff as elsewhere in this pull req.  This becomes an open invitation to new bugs, and difficult debugging sessions.

Debugging is hard. It was always hard, it will always be hard. You should always write code with the idea of debuggability in the back of your mind ... and this means, almost  always: fewer lines of code, fewer data structures, fewer if-tests.  The more that these can be reduced, the better the code.  Make the code easier to understand.

Reply:
My code is not great (see my above comment about premature optimization), but in its defense I would say it takes a bit more of error handling into account ;-)). I'll see to deduplicate it a bit though.

Reply:
I shortened the code (at the expense of not improving the error handling - I added a todo instead). It's still more verbose, but for now I would like to keep it like that and replicating the old behavior as close as possible before properly refactoring and aligning the IO code after all deprecated dependencies are gone.